crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -140,27 +136,21 @@ where
             {
                 DownloadBlock::Block(block, commitments) => break (block, commitments),
                 DownloadBlock::AtHead => {
-                    // Poll pending if it is enabled, otherwise just wait to poll head again.
-                    match pending_poll_interval {
-                        Some(interval) => {
-                            tracing::trace!("Entering pending mode");
-                            let head = head_meta
-                                .expect("Head hash should exist when entering pending mode");
-                            (next_block, next_state_update) = pending::poll_pending(
-                                tx_event.clone(),
-                                &sequencer,
-                                (head.1, head.2),
-                                interval,
-                                storage.clone(),
-                            )
-                            .await
-                            .context("Polling pending block")?;
-                        }
-                        None => {
-                            tracing::info!(poll_interval=?head_poll_interval, "At head of chain");
-                            tokio::time::sleep(head_poll_interval).await;
-                        }
-                    }
+                    const PENDING_POLL_INTERVAL: std::time::Duration =
```

Comment:
Poll pending is unsupported in p2p, so we should just sleep here for 2 secs if feature="p2p".